### DISCUSSION TITLE
How to allow non-superusers to access /users/{id} ?

### DISCUSSION THREAD

**Discussion by astroanax** [fastapi-users/fastapi-users#1408] [Q&A] [answered]
Seemingly the only way to do it is to copy a significant portion of fastapi_users.py and router/users.py in my own project. Any neater way to do it?

Reason for non-superusers to be able to access /users/{id} is so i can fetch user details like username,etc., Or should i be creating a new separate route for this?

**Comment by hgalytoby** [comment, marked as answer]
It looks like the only way to solve the issue is either to copy the code or to create a new route.
